src/charm.py

```diff
@@ -298,26 +309,14 @@ def _trace_retention_period_hours(self) -> int:
 
     def server_ca_cert(self) -> str:
         """For charm tracing."""
-        # Fixme: we do this once too many times if we're handling cert_handler.changed
-        self._update_server_ca_cert()
         return self.tempo.tls_ca_path
 
-    def _update_server_ca_cert(self) -> None:
-        """Server CA certificate for charm tracing tls, if tls is enabled."""
-        server_ca_cert = Path(self.tempo.tls_ca_path)
-        if self.coordinator.tls_available:
-            if self.coordinator.cert_handler.ca_cert:
-                server_ca_cert.parent.mkdir(parents=True, exist_ok=True)
-                server_ca_cert.write_text(self.coordinator.cert_handler.ca_cert)
-        else:  # tls unavailable: delete local cert
-            server_ca_cert.unlink(missing_ok=True)
-
     def tempo_otlp_http_endpoint(self) -> Optional[str]:
         """Endpoint at which the charm tracing information will be forwarded."""
         # the charm container and the tempo workload container have apparently the same
         # IP, so we can talk to tempo at localhost.
-        if self.coordinator and self.coordinator.tracing.is_ready():
-            return self.coordinator.tracing.get_endpoint("otlp_http")
+        if self.coordinator and self.coordinator.charm_tracing.is_ready():
```

Comment:
this is always true, or you get an AttributeError, so if you want to guard against pre-initialization access, we should use 

```
if hasattr(self, "coordinator"): ...
```